adminer.tf

```diff
@@ -0,0 +1,22 @@
+resource "aws_iam_user" "adminer" {
+  name = "adminer-${local.resource_name}"
+  tags = local.tags
+}
+
+resource "aws_iam_access_key" "adminer" {
+  user = aws_iam_user.adminer.name
+}
+
+resource "aws_iam_user_policy" "adminer" {
+  user   = aws_iam_user.adminer.name
+  policy = data.aws_iam_policy_document.adminer.json
+}
+
+data "aws_iam_policy_document" "adminer" {
```

Comment:
Here's an expanded version that handles proper permissions without knowing instance ids up front:
```
data "aws_iam_policy_document" "adminer" {
  statement {
    sid       = "ReadEnvironment"
    effect    = "Allow"
    actions   = ["elasticbeanstalk:DescribeEnvironmentResources"]
    resources = [local.env_arn]
  }

  statement {
    sid       = "EnableSSMSession"
    effect    = "Allow"
    actions   = ["ssm:StartSession"]
    resources = ["arn:aws:ssm:us-east-1::document/AWS-StartSSHSession"]
  }

  statement {
    sid       = "AllowSSMSession"
    effect    = "Allow"
    actions   = ["ssm:StartSession"]
    resources = ["arn:aws:ec2:*:*:instance/*"]

    condition {
      test     = "StringEquals"
      variable = "ec2:ResourceTag/elasticbeanstalk:application-name"
      values   = [local.app_name]
    }

    condition {
      test     = "StringEquals"
      variable = "ec2:ResourceTag/elasticbeanstalk:environment-name"
      values   = [local.env_name]
    }
  }
}
```

Reply:
To get necessary locals, I would put the following in `beanstalk.tf`. 
```

locals {
  app_name = aws_elastic_beanstalk_application.this.name
  env_name = aws_elastic_beanstalk_environment.this.name
  env_arn = "arn:aws:elasticbeanstalk:${local.region}:${local.account_id}:environment/${local.app_name}/${local.env_name}"
}
```

Reply:
This requires `local.region` and `local.account_id` which I would define in `aws.tf`:
```
locals {
  region     = data.aws_region.this.name
  account_id = data.aws_caller_identity.this.account_id
}
```

Reply:
Let's give it a shot!